acextract/AssetsCatalog.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  CatalogReader.swift
+//
+//  The MIT License (MIT)
+//
+//  Copyright (c) 2014 Bartosz Janda
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in all
+//  copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//  SOFTWARE.
+
+import Foundation
+
+enum AssetsCatalogError: ErrorType {
+    case FileDoesntExists
+    case CannotOpenAssetsCatalog
+}
+
+struct AssetsCatalog {
+    // MARK: Properties
+    let filePath: String
+    let catalog: CUICatalog
+
+    /**
+     Returns all image sets from assets catalog.
+
+     - returns: List of image sets.
+     */
+    var imageSets: [ImageSet] {
+        return catalog.allImageNames().map(imageSet(withName:))
+    }
+
+    // MARK: Initialization
+    init(path: String) throws {
+        let filePath = (path as NSString).stringByExpandingTildeInPath
```

Comment:
`String` nor `NSString` doesn't have method `expandingTildeInPath` in Swift 2 or Swift 3, or I'm not able to find one :(